cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/store/MetadataStoreTest.java

```diff
@@ -198,10 +206,16 @@ public static void setup() throws IOException {
     txManager = injector.getInstance(TransactionManager.class);
     txManager.startAndWait();
     store = injector.getInstance(MetadataStore.class);
+    auditPublisher = injector.getInstance(InMemoryAuditPublisher.class);
+    // Clear dataset creation audit messages
+    auditPublisher.popMessages();
   }
 
   @Test
   public void testPublishing() throws InterruptedException {
+    // clear audit messages
+    auditPublisher.popMessages();
```

Comment:
why is this needed twice? If its needed at the beginning of every test perhaps put it in an ``@Before`` instead of ``@BeforeClass``?